### PR TITLE
Fix #1116 - Reduce integration test running time

### LINKED ISSUE #1116: Reduce the integration test running time
### Description:
Currently the integration tests are taking some time due to building of the project per test class. If we can create a single project which can be used by multiple test classes and build it once then total build time can be reduced. 
We need to identify the test classes which can be merged to a single microgateway project and group them and build them together and run all the test classes of those projects once.

### PR DESCRIPTION
**Pull request by Rajith90**
### Purpose
Reduce the integration test running time. This will reduce the total number of project builds from 23 to 13.

### Issues

Fixes #1116 

### Automation tests
 - Unit tests added: Yes/No
 - Integration tests added: Yes/No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

File: tests/src/test/java/org/wso2/micro/gateway/tests/http2/HTTP2RequestsWithHTTP1BackEndTestCase.java [hunk 1]
```diff
@@ -21,61 +21,51 @@
...
-        info.setAuthorized(true);
-        info.setKeyType(TestConstant.KEY_TYPE_PRODUCTION);
-        info.setSubscriptionTier("Unlimited");
-
-        String configPath = "confs/http2-test.conf";
-        super.init(label, project, configPath);
-
-        jwtTokenProd = getJWT(api, application, "Unlimited", TestConstant.KEY_TYPE_PRODUCTION, 3600);
-    }
+//    @BeforeClass
```
Comment (praminda) [inline tests/src/test/java/org/wso2/micro/gateway/tests/http2/HTTP2RequestsWithHTTP1BackEndTestCase.java]: Do we need commented code? Can we remove?

File: tests/src/test/java/org/wso2/micro/gateway/tests/http2/HTTP2RequestsWithHTTP2BackEndTestCase.java [hunk 2]
```diff
@@ -132,9 +138,9 @@ public void testHTTP1RequestsViaSecureConnectionWithHTTP2BE() throws Exception {
         log.info("Response: " + response.getResponseMessage() + " , " + response.getResponseCode());
     }
 
-    @AfterClass
-    public void stop() throws Exception {
-        //Stop all the mock servers
-        super.finalize();
-    }
+//    @AfterClass
```
Comment (praminda) [inline tests/src/test/java/org/wso2/micro/gateway/tests/http2/HTTP2RequestsWithHTTP2BackEndTestCase.java]: Can we remove commented block?

File: tests/src/test/resources/testng.xml [hunk 3]
```diff
@@ -35,16 +35,9 @@
         <classes>
             <class name="org.wso2.micro.gateway.tests.apikey.APIKeyTestCase" />
             <class name="org.wso2.micro.gateway.tests.jwtRevocation.JWTRevocationSupportTestCase" />
-            <class name="org.wso2.micro.gateway.tests.security.AuthenticationFailureTestCase" />
```
Comment (praminda) [inline tests/src/test/resources/testng.xml]: Is removed test cases covered in somewhere else?

Comment (Rajith90) [reply tests/src/test/resources/testng.xml]: Yes, other classes are extending the removed ones. So only extended class name is required in the testng.xml

**Pull request merged**